src/main/cc/wfa/virtual_people/core/model/utils/distributed_consistent_hashing_test.cc

```diff
@@ -0,0 +1,272 @@
+  distribution->emplace_back(
+      std::make_pair(1, std::numeric_limits<double>::max()));
+  auto hashing_or =
+      DistributedConsistentHashing::Build(std::move(distribution));
+  EXPECT_TRUE(hashing_or.ok());
+  std::unique_ptr<DistributedConsistentHashing> hashing =
+      std::move(hashing_or.value());
+  for (int seed = 0; seed < kSeedNumber; ++seed) {
+    EXPECT_EQ(hashing->Hash(std::to_string(seed)), 1);
+  }
+}
+
+TEST(DistributedConsistentHashingTest, TestOutputDistribution_NonConsecutive) {
+  // Distribution:
+  // choice_id probability
+  // 0         0.4
+  // 2         0.2
+  // 4         0.2
+  // 6         0.2
+  std::unique_ptr<std::vector<std::pair<int32_t, double>>> distribution =
+      absl::make_unique<std::vector<std::pair<int32_t, double>>>();
+  distribution->emplace_back(std::make_pair(0, 0.4));
+  distribution->emplace_back(std::make_pair(2, 0.2));
+  distribution->emplace_back(std::make_pair(4, 0.2));
+  distribution->emplace_back(std::make_pair(6, 0.2));
+  auto hashing_or =
+      DistributedConsistentHashing::Build(std::move(distribution));
+  EXPECT_TRUE(hashing_or.ok());
+  std::unique_ptr<DistributedConsistentHashing> hashing =
+      std::move(hashing_or.value());
+  absl::flat_hash_map<int32_t, int> ouptut_counts = {
+      {0, 0},
+      {2, 0},
+      {4, 0},
+      {6, 0}};
+  for (int seed = 0; seed < kSeedNumber; ++seed) {
+    int32_t output = hashing->Hash(std::to_string(seed));
+    // The output should always be one of [0, 2, 4, 6].
+    EXPECT_TRUE(ouptut_counts.find(output) != ouptut_counts.end());
+    ++ouptut_counts[output];
+  }
+  // Absolute error more than 2% is very unlikely.
+  EXPECT_NEAR(
+      static_cast<double>(ouptut_counts[0]) / static_cast<double>(kSeedNumber),
+      0.4, 0.02);
+  EXPECT_NEAR(
+      static_cast<double>(ouptut_counts[2]) / static_cast<double>(kSeedNumber),
+      0.2, 0.02);
+  EXPECT_NEAR(
+      static_cast<double>(ouptut_counts[4]) / static_cast<double>(kSeedNumber),
+      0.2, 0.02);
+  EXPECT_NEAR(
+      static_cast<double>(ouptut_counts[6]) / static_cast<double>(kSeedNumber),
+      0.2, 0.02);
+}
+
+TEST(DistributedConsistentHashingTest, TestOutputChangeCount) {
+  // Distributions:
+  // choice_id probability_1 probability_2
+  // 0         0.4           0.2
+  // 1         0.2           0.2
+  // 2         0.2           0.2
+  // 3         0.2           0.4
+  std::unique_ptr<std::vector<std::pair<int32_t, double>>> distribution_1 =
+      absl::make_unique<std::vector<std::pair<int32_t, double>>>();
+  distribution_1->emplace_back(std::make_pair(0, 0.4));
+  distribution_1->emplace_back(std::make_pair(1, 0.2));
+  distribution_1->emplace_back(std::make_pair(2, 0.2));
+  distribution_1->emplace_back(std::make_pair(3, 0.2));
+  auto hashing_or_1 =
+      DistributedConsistentHashing::Build(std::move(distribution_1));
+  EXPECT_TRUE(hashing_or_1.ok());
+  std::unique_ptr<DistributedConsistentHashing> hashing_1 =
```

Comment:
`ASSERT_OK_AND_ASSIGN` could be useful here.

Reply:
Done.